### PR TITLE
bug: [PagoPa-2166] - fix list info for cart

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- fixed id value
- fixed payeeName e payeeTaxCode value
- fixed amount value
- fixed junit tests

#### Motivation and Context

- error reporting by users of the service

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java [hunk 1]
```diff
@@ -132,21 +131,24 @@ public static List<NoticeListItem> convertToNoticeList(TransactionListResponse t
                 .toList();
     }
 
-    public static TransactionListItem convertTransactionListItem(BizEventsViewUser viewUser, List<BizEventsViewCart> listOfCartViews) {
+    public static TransactionListItem convertTransactionListItem(BizEventsViewUser viewUser, BizEventsViewCart bizEventsViewCart, BizEventsViewGeneral bizEventsViewGeneral) {
         AtomicReference<BigDecimal> totalAmount = new AtomicReference<>(BigDecimal.ZERO);
+        BigDecimal amountExtracted = new BigDecimal(bizEventsViewCart.getAmount());
+        totalAmount.updateAndGet(v -> v.add(amountExtracted));
+        
+        /*
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java:139]: ok remove it

File: src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java [hunk 2]
```diff
@@ -171,8 +173,9 @@ private static String dateFormat(String date) {
         throw new DateTimeException("The date [" + date + "] is not in one of the expected formats " + LIST_RECEIPT_DATE_FORMAT_IN + " and cannot be parsed");
     }
 
+    /*
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java:176]: ok remove it

File: src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java [hunk 3]
```diff
@@ -132,21 +131,24 @@ public static List<NoticeListItem> convertToNoticeList(TransactionListResponse t
...
-        }
+        }*/
 
         return TransactionListItem.builder()
-                .transactionId(viewUser.getTransactionId())
-                .payeeName(listOfCartViews.size() > 1 ? payeeCartName : listOfCartViews.get(0).getPayee().getName())
-                .payeeTaxCode(listOfCartViews.size() > 1 ? "" : listOfCartViews.get(0).getPayee().getTaxCode())
-                // PAGOPA-1763: the amount value must be returned only if it is not a cart type transaction
-                .amount(listOfCartViews.size() > 1 && BooleanUtils.isTrue(viewUser.getIsDebtor()) ? null : totalAmount.get().setScale(2, RoundingMode.UNNECESSARY).toString())
+                .transactionId(viewUser.getId().replace("-p", "").replace("-d", "")) // eventId stripped of the -d or -p suffix
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java:146]: directly remove the last 2 characters.
The replace, in addition to being done twice, may not be safe if that character exists within the Id.
and just leave the comment that explains and gives context

File: src/main/java/it/gov/pagopa/bizeventsservice/service/impl/TransactionService.java [hunk 4]
```diff
@@ -75,14 +75,21 @@ public TransactionListResponse getTransactionList(String taxCode, Boolean isPaye
         }
         for (BizEventsViewUser viewUser : listOfViewUser) {
             List<BizEventsViewCart> listOfViewCart;
+            
+            String eventId = viewUser.getId().replace("-p", "").replace("-d", "");
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/bizeventsservice/service/impl/TransactionService.java:79]: see above comment

File: src/main/java/it/gov/pagopa/bizeventsservice/service/impl/TransactionService.java [hunk 5]
```diff
@@ -75,14 +75,21 @@ public TransactionListResponse getTransactionList(String taxCode, Boolean isPaye
...
+            
+            Optional<BizEventsViewCart> bizEventsViewCart = this.bizEventsViewCartRepository.findById(eventId);
+            Optional<BizEventsViewGeneral> bizEventsViewGeneral = this.bizEventsViewGeneralRepository.findById(eventId);
+            
+            /*
```
Comment (pasqualespica) [inline src/main/java/it/gov/pagopa/bizeventsservice/service/impl/TransactionService.java:84]: remove it

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/bizeventsservice/mapper/ConvertViewsToTransactionDetailResponse.java:139]: done

**Review by pasqualespica — APPROVED** [review]
🚀

**Pull request merged**